rfcs/2022-04-09-accessibility.md

```diff
@@ -0,0 +1,505 @@
+
+- [ReachUI SkipNav](https://reach.tech/skip-nav/)
+- [ChakraUI SkipNav](https://chakra-ui.com/docs/components/navigation/skip-nav)
+
+## Composite widgets
+
+*What are composite widgets?*
+
+A composite is a widget that may contain navigable descendants or childrens. Composite widgets should only have single navigation tab stop.
+Once the composite widget has focus, It's required to provide a separate navigation mechanism for users to navigate to elements that are descendants (generally with arrow keys).
+
+You can think of composite widgets as a way of grouping multiple elements into single navigatable element.
+
+Examples of composite widgets are: 
+
+- Radio Groups
+- Tabs
+- Listbox
+- Toolbars
+
+## Roving Tabindex
+
+> [Specification](https://www.w3.org/TR/wai-aria-practices-1.1/#kbd_roving_tabindex)
+
+One way to manage focus within `composite` widgets is roving tabindex pattern.  
+The element that is to be included in the tab sequence has tabindex of "0" and all other focusable elements contained in the composite have tabindex of "-1". 
+
+**Roving tabindex behaviour**: 
+
+- When the component loads, the element that will initially be included in the tab sequence will have `tabindex=0`, and all other focusable elements will have `tabindex=-1`.
+- At a time only one children will have `tabindex=0` set and all other focusable elements will have `tabindex=-1`. 
+- When the component contains focus and the user presses a navigation key that moves focus within the component (eg: arrow keys)
+  - set `tabindex=-1` on the element that has `tabindex=0`.
+  - Set `tabindex=0` on the element that will become focused as a result of the key event.
+  - Set focus with `element.focus()`, on the element that has `tabindex=0`.
+
+**Roving tabindex keyboard accessibility**
+
+- Pressing <kbd>↑</kbd> moves focus to the previous element if orientation is vertical or not defined.
+- Pressing <kbd>↓</kbd> moves focus to the next element if orientation is vertical or not defined.
+- Pressing <kbd>→</kbd> moves focus to the next element if orientation is horizontal or not defined.
+- Pressing <kbd>←</kbd> moves focus to the previous element if orientation is horizontal or not defined.
+- Pressing <kbd>Home</kbd> or <kbd>PageUp</kbd> moves focus to the first element.
+- Pressing <kbd>End</kbd> or <kbd>PageDown</kbd> moves focus to the last element.
+
+## Implementations
+
+There are various third party library implementations for roving tabindex pattern.  
+
+We tried out 3 of them to understand the pros and cons for each and decided on X.
+
+- FluentUI - [Demo](https://codesandbox.io/s/blade-comp-fluentui-focuszone-0sjyv1?file=/src/App.tsx) | [Doc](https://developer.microsoft.com/en-us/fluentui#/controls/web/focuszone) | [Pros and cons](https://codesandbox.io/s/blade-comp-fluentui-focuszone-0sjyv1?file=/src/pros-and-cons.md)
```

Comment:
What's the status of this? Have we decided on which third party library we are proceeding with? The pros and cons of whichever one we decide should come here in the rfc.

Reply:
> The pros and cons of whichever one we decide should come here in the rfc.  

yes.

We haven't decided on which library to pick up yet. Needs discussion with @kamleshchandnani @chaitanyadeorukhkar, plus the decision could be dependent on other factors like focus management & screen reader support.

Reply:
Can we add the pros and cons in collapsible section in this readme itself?